### PR TITLE
EAD-Session-5 Assignment

### PR DESCRIPTION
**Pull request by meta-manoj-sharma**

### CODE REVIEW

File: EAD-Session-5/LayeredShoppingCart/src/com/metacube/shoppingcart/ShoppingCartTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	 * test case to test user by user id 
+	 * @throws SQLException
+	 * @throws Exception
+	 */
+	@Test
+	public void testGetUserById() throws SQLException, Exception {
+
+		UserController userController = new UserController();
+
+		assertEquals("MANOJ", userController.getUserByID(2));
```
Comment (jbhavya) [inline EAD-Session-5/LayeredShoppingCart/src/com/metacube/shoppingcart/ShoppingCartTest.java]: What if the user name is not MANOJ ? The test will fail in that case.

File: EAD-Session-5/LayeredShoppingCart/src/com/metacube/shoppingcart/ShoppingCartTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	 * test case to test cart list
+	 * @throws SQLException
+	 * @throws Exception
+	 */
+	@Test
+	public void testGetCartList() throws SQLException, Exception {
+
+		CartController cartController = new CartController();
+		List<Cart> cartList = cartController.getCartList(2);
+		assertEquals("P2", cartList.get(0).getProductCode());
```
Comment (jbhavya) [inline EAD-Session-5/LayeredShoppingCart/src/com/metacube/shoppingcart/ShoppingCartTest.java]: What if user has not added P2 product in his cart?

File: EAD-Session-5/LayeredShoppingCart/src/com/metacube/shoppingcart/controller/CartController.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+public class CartController {
+	/**
+	 * @param userID for user to add product in cart
+	 * @param productID of product which is to be add in cart
+	 * @param quantity of product to be added
+	 * @throws SQLException
+	 * @throws Exception
+	 */
+	public void addProductToCart(int userID, String productID, int quantity) throws SQLException, Exception {
+		CartDao cart = new CartDao();
```
Comment (jbhavya) [inline EAD-Session-5/LayeredShoppingCart/src/com/metacube/shoppingcart/controller/CartController.java]: Follow the same practice for declaring variables of the same layer.

File: EAD-Session-5/LayeredShoppingCart/src/com/metacube/shoppingcart/ShoppingCartTest.java (hunk 1 above)
Comment (meta-manoj-sharma) [reply EAD-Session-5/LayeredShoppingCart/src/com/metacube/shoppingcart/ShoppingCartTest.java]: resolved

File: EAD-Session-5/LayeredShoppingCart/src/com/metacube/shoppingcart/ShoppingCartTest.java (hunk 2 above)
Comment (meta-manoj-sharma) [reply EAD-Session-5/LayeredShoppingCart/src/com/metacube/shoppingcart/ShoppingCartTest.java]: table logic updated

File: EAD-Session-5/LayeredShoppingCart/src/com/metacube/shoppingcart/controller/CartController.java (hunk 3 above)
Comment (meta-manoj-sharma) [reply EAD-Session-5/LayeredShoppingCart/src/com/metacube/shoppingcart/controller/CartController.java]: updated

File: EAD-Session-5/LayeredShoppingCart/src/com/metacube/shoppingcart/controller/CartController.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+		cart.executeAddProductToCartQuery(userID, productID, quantity);
+	}
+	/**
+	 * @param userID for user to update product in cart
+	 * @param productID of product which is to update add in cart
+	 * @param quantity of product to be updated
+	 * @throws SQLException
+	 * @throws Exception
+	 */
+	public void updateCartItem(int userID, String productID, int quantity) throws SQLException, Exception {
```
[the remark this answers is not in the document]
Comment (meta-manoj-sharma) [reply EAD-Session-5/LayeredShoppingCart/src/com/metacube/shoppingcart/controller/CartController.java]: updated

File: EAD-Session-5/LayeredShoppingCart/src/com/metacube/shoppingcart/controller/ProductController.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
+package com.metacube.shoppingcart.controller;
+
+import java.sql.SQLException;
+import java.util.*;
+import com.metacube.shoppingcart.dao.ProductDao;
+import com.metacube.shoppingcart.model.Product;
+	/**
```
[the remark this answers is not in the document]
Comment (meta-manoj-sharma) [reply EAD-Session-5/LayeredShoppingCart/src/com/metacube/shoppingcart/controller/ProductController.java]: updated

File: EAD-Session-5/LayeredShoppingCart/src/com/metacube/shoppingcart/controller/UserController.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+	UserDao userDaoObject = new UserDao();
+	/** 
+	 * @param user name to be add in user list
+	 * @throws SQLException
+	 * @throws Exception
+	 */
+	public void addUser(String user) throws SQLException, Exception {
+		userDaoObject.addUser(user);
+	}
+	/**
```
[the remark this answers is not in the document]
Comment (meta-manoj-sharma) [reply EAD-Session-5/LayeredShoppingCart/src/com/metacube/shoppingcart/controller/UserController.java]: updated

File: EAD-Session-5/LayeredShoppingCart/src/com/metacube/shoppingcart/view/View.java [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+	public static void main(String args[]) throws SQLException, Exception {
+		View.showMenu();
+		UserController userController = new UserController();
+		ProductController productController = new ProductController();
+		CartController cartController = new CartController();
+		Scanner input = new Scanner(System.in);
+		int menuChoice, cartChoice;
+		menuChoice = input.nextInt();
+		try {
+			while (menuChoice > 0 && menuChoice < 4) {
```
[the remark this answers is not in the document]
Comment (meta-manoj-sharma) [reply EAD-Session-5/LayeredShoppingCart/src/com/metacube/shoppingcart/view/View.java]: resolved

File: EAD-Session-5/LayeredShoppingCart/src/com/metacube/shoppingcart/view/View.java [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+									System.out.println(store.get(i).getProductCode() + " \t " + store.get(i).getProductName()
+											+ " \t " + store.get(i).getProductType() + " \t " + store.get(i).getPrice());
+								}
+								
+								System.out.println("Enter the id of the product...");
+								String productID = input.next();
+								System.out.println("Enter the quantity of the product...");
+								int productQuantity = input.nextInt();
+
+								cartController.addProductToCart(userID,productID, productQuantity);
```
[the remark this answers is not in the document]
Comment (meta-manoj-sharma) [reply EAD-Session-5/LayeredShoppingCart/src/com/metacube/shoppingcart/view/View.java]: updated

**Pull request merged**